### ISSUE TITLE
New registration is not creating a new person in the Person table

### ISSUE THREAD

**Issue by mhelali** [pickleworks/WhenWhereScheduler#81]
All Providers (and clients) and subset of Person so Person table should get filled when a new person registers.

**Comment by pickleworks** [comment]
The person table has been replaced by the asp user tables. We talked about this somewhere along the line, I think. I can still do it but I think it may be redundant .

**Comment by mhelali** [comment]
Hmmm... so my last recollection is we went from having two tables (Provider and Provider Client) that had some overlap to three tables:

The Person table would have all person's infor that is common to whether a user is a Provider Client or both a Provider and a Provider Client (that's the two possilities). This would increase our tables from two to three but no redundancy.

I'm okay with using the AspNetUsers as a replacement for the Person table but then the AspNetUsers will need to have columns added to it to be able to support what Person currently has. In general Person.Person_id was supposed to be the most central key that all others reference either directly or indirectly, so it will be a little cryptic to have that now be AspNetUsers.id. But that's the only down side, other wise it should be okay if the missing columns from Person are added to AspNetUsers. The only other concern I would have  would be this: Because this table is managed by all the automatic stuff in .net I hope it doesn't get removed or something "magically". But if we are confident system won't be doing weird auto updates to overwrite our data and that we can access the tables in the exact same way with stored procs and all as we do our own tables, this is the additional mapping fro Person to AspNetUsers that remains to be done and then we can close this issue: (basically just missing some address fields and a few extra ones for future use).

>>>> AspNetUsers has:
 Id (PK, nvarchar(128), not null)
 Email (nvarchar(256), null) 
 EmailConfirmed (bit, not null) 
 PasswordHash (nvarchar(max), null)
 SecurityStamp (nvarchar(max), null)
 PhoneNumber (nvarchar(max), null)
 PhoneNumberConfirmed (bit, not null)
 TwoFactorEnabled (bit, not null)
 LockoutEndDateUtc (datetime, null)
 LockoutEnabled (bit, not null)
 AccessFailedCount (int, not null)
 UserName (nvarchar(256), not null)

>>>> Person has:

 Person_id (PK, int, not null)
 Person_first_name (nvarchar(max), null)
 Person_last_name (nvarchar(max), null)
 Person_email (nvarchar(max), null)
 Person__password (nvarchar(max), null)
 Person_phone (nvarchar(max), null)
 Person_billing_name (nvarchar(max), null)
 Person_billing_address (nvarchar(max), null)
 Person_billing_phone (nvarchar(max), null)
 Person_misc1 (nvarchar(max), null)
 Person_misc2 (nvarchar(max), null)
 Person_misc3 (nvarchar(max), null)
 Person_misc4 (nvarchar(max), null)
 Person_misc5 (nvarchar(max), null)
 Person_tags (nvarchar(max), null)

**Comment by mhelali** [comment]
Per phone call:
We are going to try to see if we can basically merge all the three tables into two -> the current Provider table will basically be the main table that will still need to use AspUser table but will be renamed Person. The old Person and Client table will have their non-redundant fields merged into the new Person (formerly Provider table) And we will add a new field to say if the new Person has Provider privileges or not (that will be based on if they chose to become provider and they are either paying or using free trial).

So only problem I found after the call is that there definitely are things that when merged can be confusing a bit (like groups and so forth). What if we at least add a client_ or provider_ prefix for column names that ONLY apply to client or to provider?

Also to note: We need three addresses in the new mega table: billing address (currently there, optional) and business address (which will be optional)  and home address (also optional) because all three could be different (and needed differently as a client vs as a provider).

Also will need to have something like:

provider_status = 1  -- is an active provider with ability to add new events
                         =  0  -- is non-active provider (was once a provider but needs to pay be active again)
                         =  -1  -- has never been a provider so can become for whatever current free period is
                         = -2   -- was a provider but all provider data has been deleted (same as -1 would have to start from scratch again if wants to be provider but free period was used up already)

Thoughts?

**Comment by mhelali** [comment]
Another talk! Hopefully now we have it figured out! Part of the confusion was in association table which we need to rename. Going to need a client table still for all the info that might change between providers (each client has a different relationship with the provider, for example one client might owe money to one provider but not to another). BUT all the common info will still be in what is now provider table to be renamed WWUser.

== Step 1.
Rename the current Provider table to WWUser.

== Step 2.
Rename prefix of Columns from Provider to WWUser

== Step 3.
add this new column:
WWUser_provider_status = 1 -- is an active provider with ability to add new events
= 0 -- is non-active provider (was once a provider but needs to pay be active again)
= -1 -- has never been a provider so can become for whatever current free period is
= -2 -- was a provider but all provider data has been deleted (same as -1 would have to start from scratch again if wants to be provider but free period was used up already)

== Step 4.
Have 3 address fields in the Provider table:

Provider_Business_Address, Provider_Billing_Address and Provider_Home_Address (all optional and broken down into address_1, address_2, city, state, zip, phone:

WWUser__Business_address1
WWUser__Business_address2
WWUser_Business_city
WWUser_Business_state
WWUser_Business_zip
WWUser_Business_phone

WWUser_Billing_address1
WWUser_Billing_address2
WWUser_Billing_city
WWUser_Billing_state
WWUser_Billing_zip
WWUser_Billing_phone

WWUser_Home_address1
WWUser_Home_address2
WWUser_Home_city
WWUser_Home_state
WWUser_Home_zip
WWUser_Home_phone


== Step 5.
Bring over form Client table any columns that are going to be the same across the board and belong in WWUser and leave behind columns that are possibly different depending on who the client's provider is.

Move over these columns from client tab and update to new WWUser_ prefix (vs. old Provider_Client)
Provider_client_cell_phone_to_text_urgent_notification_email (nvarchar(max), null) <-- different than home/business/billing phones... basically just cell phone.

Provider_client_billing_info (nvarchar(max), null) <-- won't be necessary after Step 4 above, so remove this column

== Step 6.

Rename ProviderClient table to just client table (since we no longer use term provider as table name.) Change all column names to match the new client_ prefix.

== Step 7.
Bring over from tblProvidersClientsAndGroups into Client table these misplaced columns:

Providers_clients_associaton_and_group_notes_on_client  <-- these will be notes each provider might have on their client

rename to the correct table client_ prefix to match rest of table.

== Step 8.

tblProvidersClientsAndGroups is incorrect combination of two tables. Fix by:

Rename to simply tblClientGroupings

Will have simply the following columns:

Client_Groupings_id
Client_Groupings_provider_id
Client_Groupings_client_id_group_name

Remove the ownership_association columns, they belong to a separate table. All we need is the three columns above.

== Step 9

Add the ProviderOwnershiptbl (currently somehow attached to group table.

Will have simply the following columns:

Provider_ownership_id
Provider_ownership_owner_provider_id
Provider_ownership_slave_provider_id

== Step 10

The extra optional columns right now are only 10 char (which is useless)... make them either varchar max or at least 128 otherwise they are useless. These seems stupid but they might save us in the future! We need like 3 for WWUser, and 2 for others. Also all the key values are int (some are strings).


And then hopefully we should be all good with database!

summary:

(ASPUser+WWUser) --> all users in our database.
Client -> handle many provider to many client relationship
Client Groupings handle many provider to many clients to many groups
Provider Ownership - relationship between providers -- handle one owner (up to n owners) that own other providers and vice versa.

**Comment by mhelali** [comment]
Clarification/correction on step 8 above per talk

== Step 8.

tblProvidersClientsAndGroups is incorrect combination of two tables. Fix by:

Rename to simply tblClientGroupings

Will have simply the following columns:

Client_Groupings_id 
Client_Groupings_provider_id
Client_Groupings_client_id_group_name


I forgot to add:
Client_Groupings_client_id  <--- make sure this  client id, not WWUser id <===== IMPORTANT


The Client_Groupings_provider_id <-- make sure this is the WWUser id of the owning provider.

So table will actually have FOUR columns

Client_Groupings_id <-- just the usual primary key
Client_Groupings_provider_id  <-- WWUserid
Client_Groupings_client_id <-- Clientid
Client_Groupings_client_id_group_name  <-- group name
